app/main.py

```diff
@@ -1,34 +1,83 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self, start: Deck,
+            end: Deck,
+            is_drowned: bool = False
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = []
+        if start.row == end.row and start.column == end.column:
+            self.decks.append(Deck(start.row, start.column))
+        elif start.row == end.row:
+            for column_number in range(start.column, end.column + 1):
+                self.decks.append(Deck(start.row, column_number))
+        elif start.column == end.column:
+            for row_number in range(start.row, end.row + 1):
+                self.decks.append(Deck(row_number, start.column))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        raise ValueError("There is no such deck")
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> None:
+        deck = self.get_deck(row, column)
+        deck.is_alive = not deck.is_alive
+        is_decks_alive = []
+        for deck in self.decks:
+            is_decks_alive.append(deck.is_alive)
+        if any(is_decks_alive):
+            self.is_drowned = False
+        else:
+            self.is_drowned = True
```

Comment:
So hard. Write this inline
```suggestion
        self.is_drowned = not any(deck.is_alive for deck in self.decks)
```